src/components/Profile.vue

```diff
@@ -203,18 +224,65 @@ export default {
         backgroundImage: `url(${avatarUrl})`
       },
       certifications,
-      certKey
+      certKey,
+      errors: [],
+      invalidInputs: [],
+      msg: null
     }
   },
   methods: {
+    /**
+     * Toggle editing state.
+     * {Case A} if activeEdit === false: enter the editing state by setting activeEdit to true
+     * {Case B} if activeEdit === true: save profile changes & exit the editing state by setting activeEdit to false
+     */
     editProfile () {
-      if (this.activeEdit) {
-        UserService.setProfile(this, this.user)
-        this.editBtnMsg = 'Edit Profile'
-        this.activeEdit = false
-      } else {
+      // {Case A} Enter the editing state, then early exit
+      if (!this.activeEdit) {
```

Comment:
FYI, I rearranged the order of this function so that the quicker/shorter case is handled first and exits early. This is generally a good practice to follow for organization & readability.

Another thing that's outside this PR's scope but worth mentioning: ideally we have separate "save profile" and "edit profile" methods that are conditionally referenced in the template, i.e.
```
<button v-if="activeEdit" @click="saveProfile()">Save Profile</button>
<button v-else @click="editProfile()">Edit Profile</button>
```

No prob with the way it's set up now but imo the above style is cleaner